fetcher/linux_fetcher.go

```diff
@@ -162,7 +167,12 @@ func (lf *LinuxFetcherImpl) executeWithContext(ctx context.Context, fetcherName
 
 // executePwsh execute pwsh script by name
 func (lf *LinuxFetcherImpl) executePwsh(fetcherName string, args ...string) []byte {
-	scriptPath := config.GetBaseDir(lf.log) + "/fetch/linux/" + fetcherName
+	baseDir, err := config.GetBaseDir(lf.log)
+	if err != nil {
+		lf.log.Fatalf("Unable to get base directory: %s", err)
```

Comment:
I know this is a quite big request, but we can't fatal or panic here, we must return the error and manage it properly!